lib/src/transaction.rs

```diff
@@ -77,14 +78,14 @@ impl Transaction {
     }
 
     /// Writes the transaction to the operation store and publishes it.
-    pub fn commit(self) -> Arc<ReadonlyRepo> {
-        self.write().publish()
+    pub fn commit(self, user_settings: &UserSettings) -> Arc<ReadonlyRepo> {
+        self.write(user_settings).publish()
     }
```

Comment:
I think it's more natural to fill out these fields at the start of the transaction, i.e. `repo.start_transaction()`.
We'll need extra fields in `struct Transaction`, but user/hostname is basically the same type of the information as `description`.

Reply:
Good idea. I did that in #737. I also moved the first two commits in this PR there. I'll get back to this PR soon (I hope).